### DISCUSSION TITLE
i3 not executing bash scripts

### DISCUSSION THREAD

**Discussion by harvooo** [i3/i3#5915] [Q&A] [answered]
I have two bash scripts, one that should execute on startup;
`exec --no-startup-id ~/.config/i3/HDMI`
That contains;
```bash
 1 #!/bin/bash
 2
 3 if xrandr | grep "HDMI1 connected"; then
 4     # Execute mons -e left
 5     mons -e left
 6 fi
```

Another one that is executed upon a keybind;
`bindsym $mod+p exec --no-startup-id ~/.config/i3/screenshot`

```bash
 1 #!/bin/bash
 2
 3 temp_file=$(mktemp /tmp/screenshot.XXXXXX.png)
 4
 5 import "$temp_file"
 6
 7 xclip -selection clipboard -t image/png -i "$temp_file"
 8
 9 rm "$temp_file"
```

I have tried moving these scripts to various locations such as
/usr/bin
/usr/local/sbin
/home/USER/

I have also tried with and without the `--no-startup-id` and `--release` params but still no luck.

I have even tried combining the scripts into one exec command to look like this in my i3 config:
`bindsym $mod+Shift+p exec --no-startup-id --release "import /tmp/sc.jpg && xclip -selection clipboard -t image/png -i /tmp/sc.jpg"`

Any ideas would be much appreciated...

**Comment by yusufaktepe** [comment]
Are these scripts executable? If not, make them executable with: `chmod u+x SCRIPT_NAME`

**Comment by harvooo** [comment]
This is what my `ls -l` looks like. I have also tried it with a simple colour picker script I made with xcolor and xclip. Still no luck.
```
-rwxr-xr-x 1 harv users  300 Feb 12 16:37  screenshot
-rwxr-xr-x 1 harv users  132 Feb 12 17:23  picker
-rwxr--r-- 1 harv users  101 Feb 12 16:01  HDMI
```

**Reply by harvooo** [reply]
They run fine from terminal (and even rofi) but not under the i3 config

**Reply by yusufaktepe** [reply]
Well, I've tested your script in my config and it worked after reloading 🤷‍♂️

**Comment by orestisfl** [comment, marked as answer]
>  bindsym $mod+Shift+p exec --no-startup-id --release "import /tmp/sc.jpg && xclip -selection clipboard -t image/png -i /tmp/sc.jpg"

Correct syntax here is:
```
bindsym --release $mod+Shift+p exec --no-startup-id "import /tmp/sc.jpg && xclip -selection clipboard -t image/png -i /tmp/sc.jpg"
```

**Reply by harvooo** [reply]
A shoddy workaround was converting my scripts into single line commands just like this example which works fine. Thank you.

**Reply by orestisfl** [reply]
```
bindsym --release $mod+Shift+p exec --no-startup-id ~/path/to/script
```

should also work
